stack/stack_test.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Alex Ellis 2017. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+package stack
+
+import (
+	"testing"
+	"reflect"
+	"sort"
+)
+
+var ParseYamlTests= []struct {
+	title        string
+	filter       string
+	functions    []string
+}{
+	{
+		title:        "Filter functions only containing 'node'",
+		filter:       "node",
+		functions:    []string{"nodejs-echo"},
+	},
+	{
+		title:        "Filter functions only containing 'echo'",
+		filter:       "echo",
+		functions:    []string{"nodejs-echo", "ruby-echo"},
+	},
+	{
+		title:        "Filter functions only containing '.+-.+'",
+		filter:       ".+-[ei].+",
+		functions:    []string{"nodejs-echo", "ruby-echo", "shrink-image"},
+	},
+	{
+		title:        "Filter for all functions: '.*'",
+		filter:       ".*",
+		functions:    []string{"nodejs-echo", "ruby-echo", "shrink-image", "url-ping"},
+	},
+}
+
+func Test_parseYaml(t *testing.T) {
+	for _, test := range ParseYamlTests {
+		t.Run(test.title, func(t *testing.T) {
+			parsedYaml, err := ParseYAML("../samples.yml", test.filter)
```

Comment:
Brilliant to see the unit tests 💯 So that this test doesn't become brittle - let's create a test .yml file and keep it alongside the Golang code. Even better if you add an override which takes []bytes or string rather than a filename - then we can hard-code the test case into the file and use multiple inputs.